tests/test_framework/mod.rs

```diff
@@ -140,29 +140,30 @@ impl TestFramework {
         let taker_rpc_config = rpc_config.clone();
         let taker = Arc::new(RwLock::new(
             Taker::init(
-                Some(&temp_dir),
+                Some(temp_dir.clone().join("taker")),
                 None,
                 Some(taker_rpc_config),
                 taker_behavior.unwrap_or_default(),
                 Some(connection_type),
             )
             .unwrap(),
         ));
-
-        // Create the Makers as per given configuration map.
+        let mut base_rpc_port = 3500; // Random port for RPC connection in tests. (Not used)
+                                      // Create the Makers as per given configuration map.
         let makers = makers_config_map
             .iter()
             .map(|(port, behavior)| {
+                base_rpc_port += 1;
```

Comment:
Shouldn't we first check if the port is available for use or if it is occupied by another process?